File: Diagrammatize.Design.Pattern/code/cxx/AbstractFactory/Sample/tablefactory/TableTray.cpp
#include "TableTray.hpp"
#include <sstream>

TableTray::TableTray(const std::string& caption): Tray(caption) {
}

std::string TableTray::makeHTML() {
    std::ostringstream buffer;
    buffer << "<td>";
    buffer << "<table width=\"100%\" border=\"1\"><tr>";
    buffer << "<td bgcolor=\"#cccccc\" align=\"center\" colspan=\"" 
           << tray.size() << "\"><b>" << caption << "</b></td>";
    buffer << "</tr>\n";
    buffer << "<tr>\n";
    for (auto item : tray) {
        buffer << item->makeHTML();
    }
    buffer << "</tr></table>";
    buffer << "</td>";
    return buffer.str();
}

File: Diagrammatize.Design.Patterns/code/python/AbstractFactory/Sample/tablefactory/TableTray.py
import io
from factory import Tray

class TableTray(Tray):
    def __init__(self, caption):
        super().__init__(caption)

    def makeHTML(self):
        buffer = io.StringIO()
        buffer.write("<td>")
        buffer.write("<table width=\"100%\" border=\"1\"><tr>")
        buffer.write("<td bgcolor=\"#cccccc\" align=\"center\" colspan=\""+ str(len(self.tray)) + "\"><b>" + self.caption + "</b></td>")
        buffer.write("</tr>\n")
        buffer.write("<tr>\n")
        for item in self.tray:
            buffer.write(item.makeHTML())
        buffer.write("</tr></table>")
        buffer.write("</td>")
        return buffer.getvalue()
